File: app/src/main/java/br/com/menuvem/lojista/domain/model/TipoComponente.kt
package br.com.menuvem.lojista.domain.model

/**
 * Tipo de um [Componente] (bloco reutilizável de ficha técnica).
 * Ex.: "Pizza - Massa Grande 35cm" é MASSA; "Pizza Sabor - Calabresa" é SABOR.
 * O tipo ajuda a organizar a biblioteca de componentes e a sugerir divisões
 * (componentes de sabor são divididos por nº de sabores na mesma massa).
 */
enum class TipoComponente {
    MASSA,
    SABOR,
    EMBALAGEM,
    OUTRO
}

File: lib/domain/model/tipo_componente.dart
/// Tipo de um Componente (bloco reutilizável de ficha técnica), cadastrado
/// livremente pelo usuário — ex.: "Massa", "Sabor", "Embalagem" — para
/// organizar a biblioteca de componentes. Não é mais uma lista fixa: o
/// usuário digita o nome ao criar/editar um componente e o tipo é reutilizado
/// (por nome, sem diferenciar maiúsculas/minúsculas) nas próximas vezes.
class TipoComponente {
  final int id;
  final String nome;
  final DateTime dataCriacao;

  TipoComponente({
    this.id = 0,
    required this.nome,
    DateTime? dataCriacao,
  }) : dataCriacao = dataCriacao ?? DateTime.now();
}
